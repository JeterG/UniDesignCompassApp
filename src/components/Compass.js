import React, { Component } from 'react';
import Button from "react-bootstrap/Button";
import Layout from "../components/layout"
import "../components/bootstrap.css"
import { Row, ButtonGroup, Table } from 'react-bootstrap';
import { Col } from 'react-bootstrap';
import Amplify from 'aws-amplify';
import aws_exports from '../aws-exports'; // specify the location of aws-exports.js file on your project
import { log_list } from '../dummyData';
import Timer from 'react-compound-timer';
import { updateUser } from '../state/actions'
import { connect } from 'react-redux';
import Phase from '../components/Phase';

Amplify.configure(aws_exports);
//Comment while not using dynamic
class Compass extends Component {
    state = {
        compassName: "Universal Design Compass",
        compassPhases: [
            { key: '1', name: 'A. Define Problem', icon: '', description: '', link: '#', time: 0 },
            { key: '2', name: 'B. Research', icon: '', description: '', link: '#', time: 0 },
            { key: '3', name: 'C. Brainstorm', icon: '', description: '', link: '#', time: 0 },
            { key: '4', name: 'D. Select', icon: '', description: '', link: '#', time: 0 },
            { key: '5', name: 'E. Construct', icon: '', description: '', link: '#', time: 0 },
            { key: '6', name: 'F. Evaluate', icon: '', description: '', link: '#', time: 0 },
            { key: '7', name: 'G. Communicate', icon: '', description: '', link: '#', time: 0 },
            { key: '8', name: 'H. Redisign', icon: '', description: '', link: '#', time: 0 },
        ],
        previous: true,
        next: true,
        currentPhase: '0',
        emptyTime: "00:00:00",
        currentTime: "00:00:00",
        log: ""
    }

    componentDidMount(){
        // console.log(this.props.id)
        // if (this.props.id){
        //     getProcess(this.props.id)
        //     .then((res) => {
        //         const compass = res.data.getProcess;
        //         const compassName = compass.name;
        //         const compassPhases = compass.phaseids.items.map((phase,index) => {
        //             return { 
        //                 key: index + 1,  
        //                 name: phase.title, 
        //                 logs: phase.logs, 
        //                 description: phase.description, 
        //                 duration: phase.duration
        //             }
        //         })
        //         this.setState({
        //             compassName,
        //             // compassPhases,
        //             // previous: true,
        //             // next: true,
        //             // currentPhase: '0',
        //             // emptyTime: "00:00:00",
        //             // currentTime: "00:00:00",
        //             // log: ""
        //         })
        //     })
        //     .catch( err => {
        //         alert("there was an error with fetching your compass, we are showing you a default compass template")
        //         this.setState({
        //             compassName: "Universal Design Compass",
        //             compassPhases: [
        //                 { key: '1', name: 'A. Define Problem', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '2', name: 'B. Research', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '3', name: 'C. Brainstorm', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '4', name: 'D. Select', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '5', name: 'E. Construct', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '6', name: 'F. Evaluate', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '7', name: 'G. Communicate', icon: '', description: '', link: '#', time: 0 },
        //                 { key: '8', name: 'H. Redisign', icon: '', description: '', link: '#', time: 0 },
        //             ],
        //             previous: true,
        //             next: true,
        //             currentPhase: '0',
        //             emptyTime: "00:00:00",
        //             currentTime: "00:00:00",
        //             log: ""
        //         })
        //     })
        // } else {
            this.setState({
                compassName: "Universal Design Compass",
                compassPhases: [
                    { key: '1', name: 'A. Define Problem', icon: '', description: '', link: '#', time: 0 },
                    { key: '2', name: 'B. Research', icon: '', description: '', link: '#', time: 0 },
                    { key: '3', name: 'C. Brainstorm', icon: '', description: '', link: '#', time: 0 },
                    { key: '4', name: 'D. Select', icon: '', description: '', link: '#', time: 0 },
                    { key: '5', name: 'E. Construct', icon: '', description: '', link: '#', time: 0 },
                    { key: '6', name: 'F. Evaluate', icon: '', description: '', link: '#', time: 0 },
                    { key: '7', name: 'G. Communicate', icon: '', description: '', link: '#', time: 0 },
                    { key: '8', name: 'H. Redisign', icon: '', description: '', link: '#', time: 0 },
                ],
                previous: true,
                next: true,
                currentPhase: '0',
                emptyTime: "00:00:00",
                currentTime: "00:00:00",
                log: ""
            })
        // }
    }

    compassButtonHandler = (phase) => {//handle current phase too.
        this.setState({ currentPhase: phase.key })// Some sort of delay when logging maybe also delay in updating?
    }

    parseLogs = (logs) => {

    }

    previousButtonHandler = () => {
        var temp = !this.state.previous//need to handle active and disabled booleans too
        this.setState({ previous: temp });
    }

    nextButtonHandler = (e) => {
        var temp = !this.state.next//need to handle active and disabled booleans too
        this.setState({ next: temp });
    }

    handleTextArea = (e) => {
        this.setState({ log: e.target.value })
    }

    updateLogHandler = (event) => {
        var x = new Date()
        var minutes = x.getMinutes();
        var seconds = x.getSeconds();
        var milliseconds = x.getMilliseconds();
        var temp = minutes.toString() + ":" + seconds.toString() + ":" + milliseconds.toString()
        const log = { id: this.state.currentPhase, timestamp: temp, text: this.state.log };
        log_list.data.push(log); // Temporary
        // console.log(log_list.data); //Temporary
        this.forceUpdate();
        //API.graphql(graphqlOperation(createLog, { input: log })); //taken out temporarily!
    }

    adjustTime = (index, time) => {
        let compassPhases = this.state.compassPhases
        compassPhases[index].time = time
        this.setState({ compassPhases })
    }

    generateList(phase, currentPhase) {
        if (currentPhase === phase) {
            const filtered = log_list.data.filter((entry) => {
                return (entry.id === phase);
            })
            return (filtered.map((data) => {
                return (
                    <tbody>
                    <tr>
                    <style>{'td{background-color:rgba(50,115,220,0.3);color:grey}'}</style>
                    <td >{data.text}</td>
                    <td>{data.timestamp}</td>
                    </tr>
                    </tbody>
                );
            }));
        }
    }

    render() {
        return (
            <div className='container'>
                <h1 className='text-center'>{this.state.compassName}</h1>
                <div>
                    {
                        this.state.compassPhases.map(
                            (phase, index) => {
                                return (
                                    <Phase
                                        currentPhase={this.state.currentPhase}
                                        phase={phase}
                                        compassButtonHandler={this.compassButtonHandler}
                                        index={index}
                                        updateLogHandler={this.updateLogHandler}
                                        previous={this.state.previous}
                                        next={this.state.next}
                                        nextButtonHandler={this.nextButtonHandler}
                                        previousButtonHandler={this.previousButtonHandler}
                                        handleTextArea={this.handleTextArea}
                                        log={this.state.log}
                                        adjustTime={this.adjustTime}
                                        generateList={this.generateList}
                                        state={this.state}
                                    />

                                );
                            })
                    }
                </div>
            </div>
        );
    }
}

const mapStateToProps = ({ state }) => ({
    user: state.user
})
const mapDispatchToProps = dispatch => ({
    updateUser: (user) => dispatch(updateUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Compass);

