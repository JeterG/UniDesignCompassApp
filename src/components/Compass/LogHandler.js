import React from 'react';
import Button from "react-bootstrap/Button";
import { Row, Table } from 'react-bootstrap';
import { Col } from 'react-bootstrap';
import "../bootstrap.css"
import PropTypes from "prop-types"

const LogHandler = (props) => {
  return (
    <div>
        <Table bordered>
          {props.logs}
        </Table>
        <Row>
            <Col>
                <Button
                    className='float-left'
                    variant={props.previous ? "primary" : "secondary"}
                    onClick={() => props.previousButtonHandler()}
                    active>
                    Previous
                </Button>
            </Col>
            <Col>
                <textarea
                  placeholder="Log Text"
                  rows="5"
                  cols="20"
                  value={props.log}
                  onChange={props.handleTextArea}
                />
            </Col>
            <Col>
              <Button
                  className='float-right'
                  variant={props.next ? "primary" : "secondary"}
                  onClick={(e) => props.nextButtonHandler()}
                  active
                >
                  Next
                </Button>
            </Col>
        </Row>
        <Row>
          <Col>
            <Button
              block
              size="sm"
              variant="warning"
              onClick={(e) => props.submitLog(e,props.currentTime)}
            >
              Submit Log
            </Button>
          </Col>
        </Row>
    </div>
  );
}

export default LogHandler;

LogHandler.propTypes = {
    currentPhase :  PropTypes.number,
    phase : PropTypes.object,
    updateLogHandler: PropTypes.func,
    previous: PropTypes.bool,
    next: PropTypes.bool,
    nextButtonHandler: PropTypes.func,
    previousButtonHandler: PropTypes.func,
    handleTextArea: PropTypes.func,
    log: PropTypes.string,
    logs: PropTypes.array,
}