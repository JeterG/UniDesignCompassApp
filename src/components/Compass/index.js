import React, { Component } from 'react';
import "../bootstrap.css"
import { log_list } from '../../dummyData';
import { updateUser } from '../../state/actions'
import { connect } from 'react-redux';
import Phase from './Phase';
import {getProcess} from '../../graphql_utils/utils'

class Compass extends Component {
    state = {
        compassName: "",
        compassPhases: [],
        previous: true,
        next: true,
        currentPhase: '0',
        emptyTime: "00:00:00",
        currentTime: "00:00:00",
        log: ""
    }

    componentDidMount(){
        if (this.props.id){
            getProcess(this.props.id)
            .then((res) => {
                const compass = res.data.getProcess;
                const compassName = compass.name;
                const compassPhases = compass.phaseids.items.map((phase,index) => {
                    return { 
                        id: phase.id,
                        key: index + 1,  
                        name: phase.title, 
                        logs: phase.logs, 
                        description: phase.description, 
                        // time: phase.duration
                        time: 0
                    }
                })

                console.log(compassPhases.logs)
                this.setState({
                    compassName,
                    compassPhases,
                })
            })
            .catch( err => {
                alert("there was an error with fetching your compass, we are showing you a default compass template")
                this.setState({
                    compassName: "Universal Design Compass",
                    compassPhases: [
                        { key: '1', name: 'A. Define Problem', icon: '', description: '', link: '#', time: 0 },
                        { key: '2', name: 'B. Research', icon: '', description: '', link: '#', time: 0 },
                        { key: '3', name: 'C. Brainstorm', icon: '', description: '', link: '#', time: 0 },
                        { key: '4', name: 'D. Select', icon: '', description: '', link: '#', time: 0 },
                        { key: '5', name: 'E. Construct', icon: '', description: '', link: '#', time: 0 },
                        { key: '6', name: 'F. Evaluate', icon: '', description: '', link: '#', time: 0 },
                        { key: '7', name: 'G. Communicate', icon: '', description: '', link: '#', time: 0 },
                        { key: '8', name: 'H. Redisign', icon: '', description: '', link: '#', time: 0 },
                    ],
                    previous: true,
                    next: true,
                    currentPhase: '0',
                    emptyTime: "00:00:00",
                    currentTime: "00:00:00",
                    log: ""
                })
            })
        } else {
            this.setState({
                compassName: "There is no compass here",
            })
        }
    }

    parsePhases = () => {

    }

    parseTime

    compassButtonHandler = (phase) => {//handle current phase too.
        this.setState({ currentPhase: phase.key })// Some sort of delay when logging maybe also delay in updating?
    }
    
    previousButtonHandler = () => {
        var temp = !this.state.previous//need to handle active and disabled booleans too
        this.setState({ previous: temp });
    }

    nextButtonHandler = (e) => {
        var temp = !this.state.next//need to handle active and disabled booleans too
        this.setState({ next: temp });
    }

    handleTextArea = (e) => {
        this.setState({ log: e.target.value })
    }

    updateLogHandler = (event) => {
        var x = new Date()
        var minutes = x.getMinutes();
        var seconds = x.getSeconds();
        var milliseconds = x.getMilliseconds();
        var temp = minutes.toString() + ":" + seconds.toString() + ":" + milliseconds.toString()
        const log = { id: this.state.currentPhase, timestamp: temp, text: this.state.log };
        log_list.data.push(log); // Temporary
        // console.log(log_list.data); //Temporary
        this.forceUpdate();
        //API.graphql(graphqlOperation(createLog, { input: log })); //taken out temporarily!
    }

    adjustTime = (index, time) => {
        let compassPhases = this.state.compassPhases
        compassPhases[index].time = time
        this.setState({ compassPhases })
    }

    generateLogs(phase, currentPhase) {
        if (currentPhase === phase) {
            const filtered = log_list.data.filter((entry) => {
                return (entry.id === phase);
            })
            return (filtered.map((data,index) => {
                return (
                    <tbody key={index}>
                        <tr>
                            <style>{'td{background-color:rgba(50,115,220,0.3);color:grey}'}</style>
                            <td >{data.text}</td>
                            <td>{data.timestamp}</td>
                        </tr>
                    </tbody>
                );
            }));
        }
    }

    generatePhase = (phase,index) => (
        <Phase
            currentPhase={this.state.currentPhase}
            phase={phase}
            phaseid={phase.id}
            compassButtonHandler={this.compassButtonHandler}
            index={index}
            updateLogHandler={this.updateLogHandler}
            previous={this.state.previous}
            next={this.state.next}
            nextButtonHandler={this.nextButtonHandler}
            previousButtonHandler={this.previousButtonHandler}
            handleTextArea={this.handleTextArea}
            log={this.state.log}
            adjustTime={this.adjustTime}
            generateLogs={this.generateLogs}
        />
    )

    render() {
        const {compassName,compassPhases} = this.state
        return (
            <div className='container'>
                <h1 className='text-center'>{compassName}</h1>
                <div>
                    {compassPhases.map((phase, index) => (this.generatePhase(phase,index)))}
                </div>
            </div>
        );
    }
}

const mapStateToProps = ({ state }) => ({
    user: state.user
})
const mapDispatchToProps = dispatch => ({
    updateUser: (user) => dispatch(updateUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Compass);

