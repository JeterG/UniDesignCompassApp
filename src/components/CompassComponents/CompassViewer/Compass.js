import React,{useState, useEffect} from "react";
import ReactApexChart from 'react-apexcharts';
import { CompassWheelContainer } from "../../../styles/CompassPage"

export default ({labels, data, colors}) => {
  const [options, setOptions]= useState({})

  useEffect(() => {
    setOptions({
      chart: {
        events: {
          click: (event, chartContext, config) => console.log(event)
        }
      },
      plotOptions: {
        pie: {
          expandOnClick: false,
          donut: {
            size: '35%'
          }
        }
      },
      responsive: [{
        // breakpoint: 1130,
        // options: {
        //   dataLabels: {
        //     style: {
        //       fontSize: '8px',
        //     },
        //   },
        // }
      }],
      legend: {
        show: false
      },
      tooltip: {enabled: false},
      dataLabels: {
        enabled: true,
        formatter: (val,opts) => labels[opts.seriesIndex] ,
        style: {
          fontSize: '10px',
          color: "white",
          fontFamily: "Roboto, Helvetica Neue, sans-serif",
        },
      },
      colors
    })
  }, [labels, colors])


  return (
    <CompassWheelContainer>
      <ReactApexChart 
        options={options} 
        series={data}
        type="donut"
        height="100%"
      />
    </CompassWheelContainer>
  )
};