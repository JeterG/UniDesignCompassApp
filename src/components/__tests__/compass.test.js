import React from "react"
import renderer from "react-test-renderer"
import Login from "../../pages/Login";
import { store } from "../../state/store";
import { Provider } from 'react-redux';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Compass from '../../pages/Compass';
configure({adapter:new Adapter()});

const tree = shallow(<Compass />);




// describe("SignIn exceptions",() =>{
//      it("failed Log in",()=>{
//          console.log();
//         tree.find('[type="submit"]');
//      })
// })
