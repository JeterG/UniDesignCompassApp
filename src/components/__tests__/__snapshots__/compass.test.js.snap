// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Compass Page render 1`] = `
<Layout>
  <div
    className="container"
  >
    <h1
      className="text-center"
    >
      Universal Design Compass
    </h1>
    <div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="1"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            A. Define Problem
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="2"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            B. Research
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="3"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            C. Brainstorm
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="4"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            D. Select
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="5"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            E. Construct
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="6"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            F. Evaluate
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="7"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            G. Communicate
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="8"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            H. Redisign
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <br />
      <div>
        <Bootstrap(Row)>
          <Col
            as="div"
          >
            <Button
              active={true}
              className="float-left"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Previous
            </Button>
          </Col>
          <Col
            as="div"
          >
            <textarea
              cols="20"
              onChange={[Function]}
              placeholder="Log Text"
              rows="5"
              value=""
            />
          </Col>
          <Col
            as="div"
          >
            <Button
              active={true}
              className="float-right"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Next
            </Button>
          </Col>
        </Bootstrap(Row)>
        <Bootstrap(Row)>
          <Col
            as="div"
          />
          <Col
            as="div"
          >
            <Button
              active={false}
              block={true}
              disabled={false}
              onClick={[Function]}
              size="sm"
              type="button"
              variant="warning"
            >
              Update Log
            </Button>
          </Col>
          <Col
            as="div"
          />
        </Bootstrap(Row)>
      </div>
    </div>
  </div>
</Layout>
`;

exports[`Check Log render 1`] = `
<Layout>
  <div
    className="container"
  >
    <h1
      className="text-center"
    >
      Universal Design Compass
    </h1>
    <div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="1"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            A. Define Problem
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="2"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            B. Research
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="3"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            C. Brainstorm
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="4"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            D. Select
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="5"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            E. Construct
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="6"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            F. Evaluate
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="7"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            G. Communicate
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="8"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            H. Redisign
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <br />
      <div>
        <Bootstrap(Row)>
          <Col
            as="div"
          >
            <Button
              active={true}
              className="float-left"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Previous
            </Button>
          </Col>
          <Col
            as="div"
          >
            <textarea
              cols="20"
              onChange={[Function]}
              placeholder="Log Text"
              rows="5"
              value=""
            />
          </Col>
          <Col
            as="div"
          >
            <Button
              active={true}
              className="float-right"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Next
            </Button>
          </Col>
        </Bootstrap(Row)>
        <Bootstrap(Row)>
          <Col
            as="div"
          />
          <Col
            as="div"
          >
            <Button
              active={false}
              block={true}
              disabled={false}
              onClick={[Function]}
              size="sm"
              type="button"
              variant="warning"
            >
              Update Log
            </Button>
          </Col>
          <Col
            as="div"
          />
        </Bootstrap(Row)>
      </div>
    </div>
  </div>
</Layout>
`;

exports[`Check Timer render 1`] = `
<Layout>
  <div
    className="container"
  >
    <h1
      className="text-center"
    >
      Universal Design Compass
    </h1>
    <div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="1"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            A. Define Problem
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="2"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            B. Research
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="3"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            C. Brainstorm
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="4"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            D. Select
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="5"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            E. Construct
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="6"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            F. Evaluate
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="7"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            G. Communicate
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <div
        className="d-flex flex-column"
      >
        <ButtonGroup
          as="div"
          role="group"
          size="lg"
          toggle={false}
          vertical={false}
        >
          <Button
            active={false}
            block={true}
            bsSize="large"
            className="text-left"
            disabled={false}
            key="8"
            onClick={[Function]}
            type="button"
            variant="outline-warning"
          >
            H. Redisign
          </Button>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="outline-primary"
          >
            00:00:00
          </Button>
        </ButtonGroup>
        <div />
      </div>
      <br />
      <div>
        <Bootstrap(Row)>
          <Col
            as="div"
          >
            <Button
              active={true}
              className="float-left"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Previous
            </Button>
          </Col>
          <Col
            as="div"
          >
            <textarea
              cols="20"
              onChange={[Function]}
              placeholder="Log Text"
              rows="5"
              value=""
            />
          </Col>
          <Col
            as="div"
          >
            <Button
              active={true}
              className="float-right"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Next
            </Button>
          </Col>
        </Bootstrap(Row)>
        <Bootstrap(Row)>
          <Col
            as="div"
          />
          <Col
            as="div"
          >
            <Button
              active={false}
              block={true}
              disabled={false}
              onClick={[Function]}
              size="sm"
              type="button"
              variant="warning"
            >
              Update Log
            </Button>
          </Col>
          <Col
            as="div"
          />
        </Bootstrap(Row)>
      </div>
    </div>
  </div>
</Layout>
`;
